src/csv_reporter.cc

```diff
@@ -89,18 +89,11 @@ void CSVReporter::ReportRuns(const std::vector<Run>& reports) {
 
 void CSVReporter::PrintRunData(const Run& run) {
   std::ostream& Out = GetOutputStream();
-
-  // Field with embedded double-quote characters must be doubled and the field
-  // delimited with double-quotes.
-  std::string name = run.benchmark_name();
-  ReplaceAll(&name, "\"", "\"\"");
-  Out << '"' << name << "\",";
+  Out << '"' << StrEscape(run.benchmark_name()) << "\",";
```

Comment:
Can you please link where it's documented what should be escaped in CSV?
I'm probably looking in wrong places, https://tools.ietf.org/html/rfc4180 only says to replace `"` with `""`
?

Reply:
is there even a standard for csv?   at least \r and \n need to be escaped, \b and \f probably should be, \t is a matter of taste- the tests dont check for it in any case.  and anyway, isn't csv support being deprecated #500 ?

Reply:
> at least \r and \n need to be escaped

I'm not seeing that in a quick test with loading
> "test","my
> test"

into libreoffice:
![image](https://user-images.githubusercontent.com/88600/56228056-8fbaae00-607f-11e9-9159-b59ef9254f97.png)

And replacing `\n` with "\n":
> "test","my\ntest"

breaks it: 
![image](https://user-images.githubusercontent.com/88600/56228132-bc6ec580-607f-11e9-9723-c841a3ca268e.png)
Thus yes, i'm curious as to motivation/documentation.



> and anyway, isn't csv support being deprecated #500 ?

Yep

Reply:
you're right, \t and \\ should proabaly not be escaped for csv.  the other ones I think the "right thing" is to escape them.  \b because if you're lookign at csv, you typically want to see it on a terminal, and any \b would end up hidden, and they're more often than not in there by mistake.

Reply:
I agree that JSON changes are good.
The issue does not mention anything about CSV; are you affected by CSV side of things?

How about a middle ground solution then?
1. Click apply on that `reserve()` suggestion.
2. Drop CSV changes, thus avoiding all the questions here as to what should and should not be escaped.